File: physics/kinematic_equations/python/kinematic_equations.py
# assuming that acceleration is constant

# v -> velocity (m/s)
# v_0 -> initial velocity (m/s)
# a -> acceleration (m/s/s)
# delta_x -> change in position (m)
# t -> time (s)

# if we don't know delta_x
def v(v_0, a, t):
    return v_0 + a * t

# if we don't know velocity
def delta_x(v_0, t, a):
    return v_0 * t + 0.5 * a * t * t

# if we don't know time
def v_squared(v_0, a, delta_x):
    return v_0 * v_0 + 2 * a * delta_x

# if we don't know acceleration
def delta_x(v, v_0, t):
    return 0.5 * t * (v + v_0)

File: physics/kinematic_equations/cpp/KinematicEquations.cpp
/*
Author: xenocidewiki


It is assumed that acceleration (a) is constant.

v	=> velocity (ms^-1)
u	=> initial velocity (ms^-1)
a	=> acceleration (ms^-2)
d_x => delta x (change in position) (m)
t	=> time (s)

This is a simple application of the SUVAT equations.
*/

class KinematicEquations
{
public:

	//If d_x is unknown
	double velocity(double u, double a, double t) {
		return u + a * t;
	}

	//If v is unknown
	double distance_from_acceleration(double u, double a, double t) {
		return u * t + 0.5 * (a * t * t);
	}

	//If t is unknown
	double velocity_squared(double u, double a, double d_x) {
		return u*u + (2 * a * d_x);
	}

	//If a is unknown
	double distance_from_velocity(double u, double v, double t) {
		return ((v + u) / 2) * t;
	}
};
